### PR TITLE
Check for same submission

### PR DESCRIPTION
**Pull request by matthewrsj**

### CODE REVIEW

File: request_handler.py [hunk 1]
```diff
@@ -169,6 +176,45 @@ def is_rate_violation(self):
             self.reset_rate()
         return False
 
+    def is_duplicate(self, submission):
+        """Calculates a hash from a submission and adds it to the hash list"""
+        sub_hash = hashlib.sha512()
+        sub_hash.update(str(submission))
+        sub_hash = sub_hash.hexdigest()
```
Comment (Kennric) [inline request_handler.py]: this looks a little odd - you are setting sub_hash to a new hash, updating that hash with the submission, then setting sub_hash to the hexdigest of sub_hash - maybe the digest should be in a different variable for clarity?

File: request_handler.py [hunk 2]
```diff
@@ -169,6 +176,45 @@ def is_rate_violation(self):
...
+        if not self.check_time_diff_hash():
+            return self.check_for_duplicate_hash(sub_hash)
+        return False
+
+    def check_time_diff_hash(self):
+        """Checks time_diff_hash for a value greater than one hour"""
+        self.set_time_diff_hash()
+        if self.time_diff_hash > (3600):  # 60 seconds * 60 minutes
```
Comment (Kennric) [inline request_handler.py]: if we put the duplicate timeframe into settings, like ceiling, then we can tune it later in production

File: request_handler.py (hunk 1 above)
Comment (matthewrsj) [inline request_handler.py]: @Kennric I can do that. I tried chaining the methods originally but that broke everything.

File: request_handler.py [hunk 3]
```diff
@@ -169,6 +176,45 @@ def is_rate_violation(self):
...
+        self.hash_list = []
+        self.time_diff_hash = 0
+        self.start_time_hash = datetime.now()
+
+    def check_for_duplicate_hash(self, sub_hash):
+        """
+        Checks for a duplicate hash in hash_list
+        Returns True if there is a duplicate and False otherwise
+        """
+        for index, element in enumerate(self.hash_list):
```
Comment (Kennric) [inline request_handler.py]: You should be able to check this without a loop, look at python list methods

File: docs/source/usage.rst [hunk 4]
```diff
@@ -11,12 +11,15 @@ In conf.py change the following variables to match your setup:
     EMAIL = u'email@example.com'
     TOKN = u'1234567890-='
     CEILING = 10
+    DUPL_CHECK_LIM = 3600
```
Comment (Kennric) [inline docs/source/usage.rst]: Didn't you just change the name of this in the conf file itself?

**Comment by matthewrsj** [comment]
@Kennric I think I noticed that the same time you did, changed with https://github.com/osuosl/formsender/commit/dac174d054e27dab1c38441bee5c32c1d75662b2

**Comment by Kennric** [comment]
+1

**Pull request merged**